### PR TITLE
Repository checksums

### PR DESCRIPTION
**Pull request by g4s8**
#117 - added `RepositoryChecksums` to read checksum files to add values to response headers then.

### CODE REVIEW

File: src/main/java/com/artipie/maven/repository/RepositoryChecksums.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        return rxsto.list(artifact).flatMapObservable(Observable::fromIterable)
+            .filter(key -> SUPPORTED_ALGS.contains(extension(key)))
+            .flatMapSingle(
+                item -> rxsto.value(item)
+                    .flatMap(pub -> new Concatenation(pub).single())
+                    .map(buf -> new Remaining(buf).bytes())
+                    .map(
+                        bytes -> new MapEntry<>(
+                            extension(item), new String(bytes, StandardCharsets.UTF_8)
+                        )
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/maven/repository/RepositoryChecksums.java:86]: @g4s8 I'd suggest to simplify this code using `PublisherAs` from asto

File: src/main/java/com/artipie/maven/repository/RepositoryChecksums.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    /**
+     * Key extension.
+     * @param key Key
+     * @return Extension string
+     */
+    private static String extension(final Key key) {
+        final String src = key.string();
+        return src.substring(src.lastIndexOf('.') + 1).toLowerCase(Locale.US);
+    }
```
Comment (olegmoz) [inline src/main/java/com/artipie/maven/repository/RepositoryChecksums.java:106]: @g4s8 there is `KeyLastPart` class in ASTO that seems to be the thing you need here

File: src/main/java/com/artipie/maven/repository/RepositoryChecksums.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            .filter(key -> SUPPORTED_ALGS.contains(extension(key)))
+            .flatMapSingle(
+                item -> rxsto.value(item)
+                    .flatMap(pub -> new Concatenation(pub).single())
+                    .map(buf -> new Remaining(buf).bytes())
+                    .map(
+                        bytes -> new MapEntry<>(
+                            extension(item), new String(bytes, StandardCharsets.UTF_8)
+                        )
+                    )
```
Comment (olegmoz) [inline src/main/java/com/artipie/maven/repository/RepositoryChecksums.java:87]: @g4s8 you may use `PublisherAs` class here to read a string from `Publisher`

File: src/main/java/com/artipie/maven/repository/RepositoryChecksums.java (hunk 2 above)
Comment (g4s8) [reply src/main/java/com/artipie/maven/repository/RepositoryChecksums.java:99]: @olegmoz `KeyLastPart` is a bit different: it takes last part after splitting by `/` char, but I need something like file extendsion - the last part after dot `.`

**Comment by g4s8** [comment]
@olenagerasimova @olegmoz thanks, fixed and answered

**Comment by 0crat** [comment]
Job `gh:artipie/maven-adapter#119` is not assigned, can't get performer

**Pull request merged**